REGRESSION(286686@main?): [macOS] http/tests/workers/service/shownotification-allowed.html is a constant failure (flaky in EWS)

youennf opened the pull request:
#### 11f3d0601ecc98aa0b9879e63a5f655ef472326b
<pre>
REGRESSION(286686@main?): [macOS] http/tests/workers/service/shownotification-allowed.html is a constant failure (flaky in EWS)
<a href="https://rdar.apple.com/140875944">rdar://140875944</a>
<a href="https://bugs.webkit.org/show_bug.cgi?id=283999">https://bugs.webkit.org/show_bug.cgi?id=283999</a>

Reviewed by Chris Dumez.

http/tests/workers/service/shownotification-allowed.html was failing as we were:
- stopping the service worker due to network process crash
- recreating the service worker, but in a different process from the web page
- sending a message to the web page via self.clients
- But the web page is not reregistering it as a service worker client since it does not create a new connection to the network process.

To fix this, we make sure the web page recreates a connection to the network process and we make sure the service worker loops until it can send the message to at least one client.

* LayoutTests/http/tests/workers/service/resources/shownotification-worker.js:
(let.messageClients):
* LayoutTests/http/tests/workers/service/shownotification-allowed.html:
* LayoutTests/platform/mac-wk2/TestExpectations:

Canonical link: <a href="https://commits.webkit.org/287520@main">https://commits.webkit.org/287520@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/f740e601d919d9f2097a15852651cbbd47b84bfd

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/79982 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/58979 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/33380 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/84496 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/30956 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/68043 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/7275 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/62516 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/20342 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/83049 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/47/builds/52580 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/72843 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/42825 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/49918 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/26996 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/29418 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/71038 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/27487 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/85928 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/7198 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/5065 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/70789 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/7375 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/68684 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/70033 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/17441 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/14030 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/12965 "Passed tests") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/7162 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/7009 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/10521 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/8813 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aproskuryakov commented on LayoutTests/platform/mac-wk2/TestExpectations:
```diff
@@ -1346,6 +1346,7 @@ webkit.org/b/241265 [ Debug ] imported/w3c/web-platform-tests/html/rendering/rep
 
 # NOTIFICATION_EVENT is Ventura+
 [ Ventura+ ] http/tests/workers/service/shownotification-allowed-document.html [ Pass ]
+[ Ventura+ ] http/tests/workers/service/shownotification-allowed.html [ Pass ]
```
Ventura is the oldest supported macOS.

youennf replied:
I removed Ventura+ from these expectations.

youennf committed:
REGRESSION(286686@main?): [macOS] http/tests/workers/service/shownotification-allowed.html is a constant failure (flaky in EWS)
rdar://140875944
https://bugs.webkit.org/show_bug.cgi?id=283999

Reviewed by Chris Dumez.

http/tests/workers/service/shownotification-allowed.html was failing as we were:
- stopping the service worker due to network process crash
- recreating the service worker, but in a different process from the web page
- sending a message to the web page via self.clients
- But the web page is not reregistering it as a service worker client since it does not create a new connection to the network process.

To fix this, we make sure the web page recreates a connection to the network process and we make sure the service worker loops until it can send the message to at least one client.

* LayoutTests/http/tests/workers/service/resources/shownotification-worker.js:
(let.messageClients):
* LayoutTests/http/tests/workers/service/shownotification-allowed.html:
* LayoutTests/platform/mac-wk2/TestExpectations:

Canonical link: https://commits.webkit.org/287520@main
